org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/IBuildSupport.java

```diff
@@ -70,13 +72,18 @@ default void refresh(IResource resource, CHANGE_TYPE changeType, IProgressMonito
 			return;
 		}
 		if (changeType == CHANGE_TYPE.DELETED) {
+			if (IJavaProject.CLASSPATH_FILE_NAME.equals(resource.getName())) {
+				IProject project = resource.getProject();
+				ProjectUtils.removeJavaNatureAndBuilder(project, monitor);
```

Comment:
what about classpath file removed before staring jdt.ls?

Reply:
See case 2.

Reply:
the problem affects gradle users (and maven)

Reply:
Yes, but the Java builder causes the issue when there isn't .classpath. When we importing a maven or gradle project, we create .classpath (and .project).